### PR TITLE
Add GitLab integration functionality

### PR DESCRIPTION
**Pull request by e-for-eshaan**
https://github.com/middlewarehq/middleware/issues/455

This pull request adds functionality for integrating with GitLab. It includes the following changes:

- Creates new functions for GitLab Personal Access Token (PAT) validity and missing scopes

- Adds `meta` argument in `linkProvider` function to store custom domain for GitLab tokens

- Updates the integration API to handle `provider_meta`

- Refactors the GitHub integration card into a separate component

- Refactors the modal body into a separate component

- Adds methods for linking and unlinking GitLab

- Creates an integration card and integration modal for the GitLab PAT

- Renders the integration card for GitLab

Please review and merge this pull request to enable GitLab integration functionality.

https://github.com/user-attachments/assets/d8b14dd4-e29d-479e-80fe-3f691455dd45

image

### Updated colors
https://github.com/user-attachments/assets/8c2c9fdf-2df0-4f68-980f-fd1789add368

### CODE REVIEW

File: web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      {
+        height: '32px',
+        top: '360px'
+      },
+      {
+        height: '32px',
+        top: '425px'
+      }
+    ].map((item) => ({ ...item, ...baseStyles }));
+  }, []);
```
Comment (shivam-bit) [inline web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx:292]: Please explain this

File: web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        top: '378px'
+      },
+      {
+        height: '120px',
+
+        top: '806px'
+      }
+    ].map((item) => ({ ...item, ...baseStyles }));
+  }, []);
+
```
Comment (shivam-bit) [inline web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx:223]: seems repeated

File: web-server/public/assets/gitlabPAT.png
Comment (shivam-bit) [inline web-server/public/assets/gitlabPAT.png:1]: I think gitlab also supports dark mode
Please use that image if possible

**Review by shivam-bit — CHANGES_REQUESTED** [review]

File: web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx (hunk 1 above)
Comment (e-for-eshaan) [reply web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx:292]: the squares that highlight the necessary permissions

image

File: web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx (hunk 2 above)
Comment (e-for-eshaan) [reply web-server/src/content/Dashboards/ConfigureGithubModalBody.tsx:223]: Naah, base styles are slightly different, refactor would prolly just make it more messy.

File: web-server/public/assets/gitlabPAT.png
Comment (e-for-eshaan) [reply web-server/public/assets/gitlabPAT.png:1]: image

File: web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx (hunk 1 above)
Comment (shivam-bit) [reply web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx:292]: why we need css for that? are we expanding the images when user hover the mouse on the image

Comment (shivam-bit) [reply web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx:292]: Approving it anyhow, but why not we are simply adding the red boxes in the image itself .

**Review by shivam-bit — DISMISSED** [review]

**Review by jayantbh — APPROVED** [review]

**Pull request merged**